app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt

```diff
@@ -55,18 +54,51 @@ class RecentTransactionsFragment : BaseFragment(), RecentTransactionsView, OnRef
         savedInstanceState: Bundle?,
     ): View {
         _binding = FragmentRecentTransactionsBinding.inflate(inflater, container, false)
-        recentTransactionsPresenter?.attachView(this)
         sweetUIErrorHandler = SweetUIErrorHandler(activity, binding.root)
         showUserInterface()
         setToolbarTitle(getString(R.string.recent_transactions))
+        viewModel =
+            ViewModelProviders.of(this, viewModelFactory)[RecentTransactionViewModel::class.java]
         if (savedInstanceState == null) {
-            recentTransactionsPresenter?.loadRecentTransactions(false, 0)
+            viewModel.loadRecentTransactions(false, 0)
         }
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        viewModel.transactionLoadingStateLiveData.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                is TransactionLoadingState.ShowProgress -> {
+                    showProgress()
+                }
+                is TransactionLoadingState.HideProgress -> {
+                    hideProgress()
+                }
+                is TransactionLoadingState.RecentTransactions -> {
+                    hideProgress()
+                    val transactions: List<Transaction> = state.transactions
```

Comment:
Do we need to create the transactions variable here? What about ` showRecentTransactions(state.transactions)` ?

Reply:
yes we dont need to . I will wrap it up in a single line then